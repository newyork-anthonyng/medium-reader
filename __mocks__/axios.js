/* global jest */
const mockAxios = {
  get: jest.fn(() => {
    return Promise.resolve({})
  })
}

export default mockAxios
