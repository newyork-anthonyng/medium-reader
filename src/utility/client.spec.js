/* global jest, describe, it, expect */
import getMediumText from './client'
import axios from 'axios'

describe('getMediumText', () => {
  it('should call correct link', () => {
    const mediumLink = 'someMediumLink.com'

    return getMediumText(mediumLink).then(() => {
      expect(axios.get).toHaveBeenCalledTimes(1)
      expect(axios.get.mock.calls[0][0]).toMatchSnapshot()
    })
  })

  it('should return Promise', () => {
    const mediumLink = 'someMediumLink.com'

    const result = getMediumText(mediumLink)
    expect(typeof result.then).toEqual('function')
    return result
  })

  it('should return correct response information', () => {
    const mediumLink = 'someMediumLink.com'
    axios.get = jest.fn(() => {
      return Promise.resolve({
        data: {
          data: 'This is some text from medium.com'
        }
      })
    })

    return getMediumText(mediumLink).then((data) => {
      expect(data).toMatchSnapshot()
    })
  })

  it('should handle rejection gracefully', () => {
    const mediumLink = 'someMediumLink.com'
    axios.get = jest.fn(() => {
      return Promise.reject(new Error('Uh oh, something happened'))
    })

    const result = getMediumText(mediumLink)
    expect(typeof result.then).toEqual('function')
  })
})
