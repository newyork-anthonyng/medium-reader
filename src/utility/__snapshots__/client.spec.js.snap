// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getMediumText should call correct link 1`] = `"http://localhost:3000/medium?mediumLink=someMediumLink.com"`;

exports[`getMediumText should return correct response information 1`] = `"This is some text from medium.com"`;
